packages/jest-matchers/src/index.js

```diff
@@ -57,7 +58,17 @@ const makeThrowingMatcher = (
   actual: any,
 ): ThrowingMatcherFn => {
   return function throwingMatcher(expected, ...rest) {
-    const matcherContext: MatcherContext = {isNot};
+    let dontThrow = false;
+    const matcherContext: MatcherContext = Object.assign(
+      // dontThrow will prevent the matcher from throwing errors during
+      // the test execution and add them to GLOBAL_STATE.state.notThrownErrors
+      // instead. Snapshot matcher uses it, because we want to log **all**
+      // snapshot failures in the test, and if a matcher throws, the test
+      // execution is stopped immediately.
```

Comment:
" When `throws` is disabled, the matcher will not throw errors during test execution but instead add them to the global matcher state. If a matcher throws, test execution is normally stopped immediately. The snapshot matcher uses it because we want to log *all* snapshot failures in a test."